### PR TITLE
fix the history function and man page

### PR DESCRIPTION
**Pull request by krader1961**
The previous commit to add a `--with-timestamp` flag to the `history` command
caused me to notice the history function didn't recognize the new long option.
Neither did it recognize the short options for the builtin command. This
change fixes both of those issues.

### CODE REVIEW

**Comment by krader1961** [comment]
@zanchey Please take a look and comment on whether I've introduced more problems than I've solved with this change.

Obviously I welcome feedback from everyone so please comment if you have an issue with this change.

File: doc_src/history.txt [hunk 1]
```diff
@@ -2,15 +2,21 @@
 
 \subsection history-synopsis Synopsis
 \fish{synopsis}
-history ( --merge | --save | --clear )
-history ( --search | --delete ) [ --prefix "prefix string" | --contains "search string" ]
+history ( -m | --merge )
+history ( -s | --save )
+history ( -l | --clear )
+history ( -s | --search ) [ -t | --with-time ] [ -p "prefix string" | --prefix "prefix string" | -c "serach string | --contains "search string" ]
+history ( -d | --delete ) [ -t | --with-time ] [ -p "prefix string" | --prefix "prefix string" | -c "serach string | --contains "search string" ]
```
Comment (kopischke) [inline doc_src/history.txt]: s/serach/search (this line and previous)

Comment (krader1961) [inline doc_src/history.txt]: fixed

**Comment by krader1961** [comment]
Merged as commit cbee315b1bd674fdef1937fa95022ef830e565ce.

File: share/functions/history.fish [hunk 2]
```diff
@@ -1,61 +1,61 @@
...
-                case "-*" "--*"
-                    printf ( _ "%s: invalid option -- %s\n" ) history $argv[1] >&2
-                    return 1
-            end
-        end
-    else
-        # Execute history builtin without any argument.
+function history --shadow-builtin --description "display or manipulate interactive command history"
+    if not set -q argv[1]
+        # No arguments so execute history builtin using it's default behavior to display the entire
```
Comment (zanchey) [inline share/functions/history.fish]: s/it's/its/

**Comment by zanchey** [comment]
Looks fine, glad you've merged this!

**Pull request closed without merging**